runtime/pendulum/src/zenlink.rs

```diff
@@ -122,41 +122,7 @@ where
 	}
 }
 
-impl TryFrom<CurrencyId> for ZenlinkAssetId {
-	type Error = ();
-
-	fn try_from(currency_id: CurrencyId) -> Result<Self, Self::Error> {
-		let para_chain_id: u32 = ParachainInfo::parachain_id().into();
-		match currency_id {
-			CurrencyId::Native =>
-				Ok(ZenlinkAssetId { chain_id: para_chain_id, asset_type: NATIVE, asset_index: 0 }),
-			CurrencyId::XCM(xcm) => Ok(ZenlinkAssetId {
-				chain_id: para_chain_id,
-				asset_type: LOCAL,
-				asset_index: xcm as u64,
-			}),
-		}
-	}
-}
-
-impl TryFrom<ZenlinkAssetId> for CurrencyId {
-	type Error = ();
-	fn try_from(asset_id: ZenlinkAssetId) -> Result<Self, Self::Error> {
-		let para_chain_id: u32 = ParachainInfo::parachain_id().into();
-		if asset_id.chain_id != para_chain_id {
-			return Err(())
-		}
-
-		match asset_id.asset_type {
-			NATIVE => Ok(CurrencyId::Native),
-			LOCAL => {
-				let foreign_currency_id_option = asset_id.asset_index.try_into();
-				match foreign_currency_id_option {
-					Ok(foreign_currency_id) => Ok(CurrencyId::XCM(foreign_currency_id)),
-					Err(_) => Err(()),
-				}
-			},
-			_ => Err(()),
-		}
-	}
+// This will be replaced with PR 241
+pub fn zenlink_id_to_currency_id(asset_id: ZenlinkAssetId) -> Result<CurrencyId, ()> {
```

Comment:
zenlink_id_to_currency_id will be replaced with [this](https://github.com/pendulum-chain/pendulum/blob/b58616fb38a94ab4afa23a659e9dfafeed60f729/runtime/common/src/zenlink.rs#L28C1-L61C2) implementation. It might be a good idea to merge the [other PR](https://github.com/pendulum-chain/pendulum/pull/246) first and then merge it into the current branch, allowing me to replace the function.

Alternatively, we can merge it as is, but this function is quite ugly to reach the main branch :joy:

Reply:
I think we can live with this useless dummy implementation for now since we will immediately change it in #241 as you mentioned. Merging this PR first and then integrating the changes to #241 is the better order IMO.